lib/encoder.js

```diff
@@ -695,12 +707,17 @@ function defineInstanceMembers() {
     }
     return value;
   };
-  this.encodeCustom = function (value, name) {
-    const handler = customEncoders[name];
+  this.encodeCustom = function (value, customTypeName) {
+
+    // Special handling for vector custom types (since they have args)
+    if (customTypeName.indexOf(customTypeNames.vector) >= 0) {
```

Comment:
why does this one use `indexOf` instread of `startsWith`? If theres a good reason please add a comment

Reply:
Nope, you're absolutely right @hhughes .  Fixing.